modules/zpca/zpca.py

```diff
@@ -111,22 +150,42 @@ def parse_scree_logs(self, f: dict):
                 words.append(word)
                 word = []
 
-        # Concatinating each character
+        # Concatinating each word
         for k in words:
-            listToStr1.append("".join([str(elem) for elem in k]))
-        # Conactinating each word
-        for k in listToStr1:
             listToStr.append("".join([str(elem) for elem in k]))
 
-        exp_car_str = "Percentage of Explained Variance"
+        len_listToStr = len(listToStr)
+
+        if len_listToStr == 8:
+            return self._parse_scree_helper_3(listToStr)
+        elif len_listToStr == 6:
+            return self._parse_scree_helper_2(listToStr)
+        else:
+            return {}
```

Comment:
I generally do not understand the parsing logic with these `words` vs. `characters`, `listToStr` len being `6` or `8`. Could you please add a little explanation to the code (in a comment)?